InnerEye/ML/utils/checkpoint_handling.py

```diff
@@ -79,26 +80,18 @@ def additional_training_done(self) -> None:
 
     def get_recovery_path_train(self) -> Optional[Path]:
         """
-        Decides the checkpoint path to use for the current training run. If a run recovery object is used, use the
-        checkpoint from there, otherwise use the checkpoints from the current run.
+        Decides the checkpoint path to use for the current training run. Looks for the latest checkpoint in the
+        checkpoint folder. If run_recovery is provided, the checkpoints will have been downloaded to this folder
+        prior to calling this function. Else, if the run gets pre-empted and automatically restarted in AML,
+        the latest checkpoint will be present in this folder too.
         :return: Constructed checkpoint path to recover from.
         """
-        start_epoch = self.container.start_epoch
-        if start_epoch > 0 and not self.run_recovery:
-            raise ValueError("Start epoch is > 0, but no run recovery object has been provided to resume training.")
+        recovery = find_recovery_checkpoint_and_epoch(self.container.checkpoint_folder)
+        if recovery is not None:
+            local_recovery_path, recovery_epoch = recovery
+            self.container._start_epoch = recovery_epoch
```

Comment:
If we need that field from the outside of the class, it should not be private. recovery_start_epoch?

Reply:
true